### PR TITLE
Add isNull and isNotNull filter relations

### PR DESCRIPTION
**Pull request by baiqiushi**
Solve issue #464 
In addition to current [Cloudberry Query Model](https://docs.google.com/document/d/1-xTBxdIg_y_nHMpztZdEG3i_IOpeKhwIT6UQJVDL0UM/edit?usp=sharing), add 2 `filter` predicates.

- `IS NULL`
  * **filter grammer**: `isNull`
  * **Applied DataType**: `Boolean`, `Number`, `Point`, `Time`, `String`, `Text`
  * **Implementation**:
      -- SQL++ = `IS UNKNOWN`
      -- MySQL = `IS NULL`
- `IS NOT NULL`
  * **filter grammer**: `isNotNull`
  * **Applied DataType**: `Boolean`, `Number`, `Point`, `Time`, `String`, `Text`
  * **Implementation**:
      -- SQL++ = `IS NOT UNKNOWN`
      -- MySQL = `IS NOT NULL`

**Example**:
```
"filter": [
   {
     "field": "fieldP",
     "apply": "func",
     "relation": "isNull|isNotNull",
     "value": []
   }
]
```
**NOTE: `No implementation for another 2 QueryGenerators: PostgreSQL & SparkSQL`**

### CODE REVIEW

File: cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AsterixQueryGenerator.scala [hunk 1]
```diff
@@ -183,6 +183,12 @@ abstract class AsterixQueryGenerator extends IQLGenerator {
       case Relation.inRange => {
         s"$fieldExpr >= ${typeImpl.datetime}('${filter.values(0)}') and $fieldExpr < ${typeImpl.datetime}('${filter.values(1)}')"
       }
+      case Relation.isNull => {
+        s"$fieldExpr is unknown"
+      }
+      case Relation.isNotNull => {
+        s"$fieldExpr is not unknown"
+      }
```
Comment (JavierJia) [inline cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AsterixQueryGenerator.scala]: Is it better to move the repeated code from line 186 to 191 in different functions to a unified place? E.g., could we handle the `isNull` or `isNotNull` in the entry caller `parseFilterRelation`?

File: cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/QueryValidator.scala [hunk 2]
```diff
@@ -169,6 +181,9 @@ object QueryValidator {
   }
 
   def validateTextRelation(relation: Relation, values: Seq[Any]): Unit = {
+    if (relation == Relation.isNull || relation == Relation.isNotNull) {
+      return
+    }
```
Comment (JavierJia) [inline cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/QueryValidator.scala]: Similar to the previous comments, if the `null` logic is handled in a unified place, we should not need to validate it here.

**Comment by chenlica** [comment]
@JavierJia Thanks for giving a detailed review for the first scala PR of @baiqiushi :-)

File: cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AsterixQueryGenerator.scala (hunk 1 above)
Comment (baiqiushi) [reply cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AsterixQueryGenerator.scala]: @JavierJia Sorry, what do you mean by `JSONRequest` changes? Do you mean `JSONParser`?
For the unification part, I agree, I just didn't know how to do it without changing too much code, I will try to think of one way. Thank you.

**Comment by chenlica** [comment]
I saw some changes that are simply related to reformatting.  It will be good to separate such changes from logical changes.  We can do them as a separate PR.  Just some thoughts ...

**Comment by baiqiushi** [comment]
Yeah, there are only 2 empty lines which are modified because that function was in my earlier commit, but after taking Jianfeng's advice to unify the logic to a higher place, that function was reverted to original version, those two empty lines were only reverted because it's more coherent to the format of other original code, so I left them there.

**Pull request merged**